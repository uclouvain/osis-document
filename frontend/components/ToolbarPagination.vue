<!--
  -
  - OSIS stands for Open Student Information System. It's an application
  - designed to manage the core business of higher education institutions,
  - such as universities, faculties, institutes and professional schools.
  - The core business involves the administration of students, teachers,
  - courses, programs and so on.
  -
  - Copyright (C) 2015-2023 UniversitÃ© catholique de Louvain (http://www.uclouvain.be)
  -
  - This program is free software: you can redistribute it and/or modify
  - it under the terms of the GNU General Public License as published by
  - the Free Software Foundation, either version 3 of the License, or
  - (at your option) any later version.
  -
  - This program is distributed in the hope that it will be useful,
  - but WITHOUT ANY WARRANTY; without even the implied warranty of
  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  - GNU General Public License for more details.
  -
  - A copy of this license - GNU General Public License - is available
  - at the root of the source code of this program.  If not,
  - see http://www.gnu.org/licenses/.
  -
  -->

<template>
  <div class="btn-group">
    <button
        class="btn btn-default"
        type="button"
        :class="{'disabled': currentPage == 1}"
        @click="currentPage != 1 && $emit('onChangeCurrentPage', currentPage - 1)"
    >
      <span class="fas fa-angle-up" />
    </button>
    <a
        class="btn btn-default disabled"
        role="button"
    >
      {{ $t('editor.pagination', {currentPage, pages}) }}
    </a>
    <button
        class="btn btn-default"
        type="button"
        :class="{'disabled': currentPage == pages}"
        @click="currentPage != pages && $emit('onChangeCurrentPage', currentPage + 1)"
    >
      <span class="fas fa-angle-down" />
    </button>
  </div>
</template>

<script lang="ts" setup>
defineProps<{
  currentPage: number,
  pages: number,
}>();

defineEmits<{
  (e: 'onChangeCurrentPage', page: number): void
}>();
</script>
