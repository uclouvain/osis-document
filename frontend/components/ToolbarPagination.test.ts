/*
 *
 * OSIS stands for Open Student Information System. It's an application
 * designed to manage the core business of higher education institutions,
 * such as universities, faculties, institutes and professional schools.
 * The core business involves the administration of students, teachers,
 * courses, programs and so on.
 *
 * Copyright (C) 2015-2023 UniversitÃ© catholique de Louvain (http://www.uclouvain.be)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * A copy of this license - GNU General Public License - is available
 * at the root of the source code of this program.  If not,
 * see http://www.gnu.org/licenses/.
 *
 */

import {it, expect} from 'vitest';
import {mount} from "@vue/test-utils";
import ToolbarPagination from "./ToolbarPagination.vue";

const props = {
  currentPage: 1,
  pages: 12,
};

it('should mount', () => {
  const wrapper = mount(ToolbarPagination, {props});
  expect(wrapper.html()).toMatchSnapshot();
});

it('should not emit on disabled previous button', async () => {
  const wrapper = mount(ToolbarPagination, {props});
  await wrapper.find('button.disabled').trigger('click');
  expect(wrapper.emitted('onChangeCurrentPage')).toBeFalsy();
});

it('should not emit on disabled next button', async () => {
  const wrapper = mount(ToolbarPagination, {
    props: {
      ...props,
      currentPage: 12,
    },
  });
  await wrapper.find('button.disabled').trigger('click');
  expect(wrapper.emitted('onChangeCurrentPage')).toBeFalsy();
});

it('should emit on next button', async () => {
  const wrapper = mount(ToolbarPagination, {props});
  await wrapper.findAll('button')[1].trigger('click');
  expect(wrapper.emitted()).toHaveProperty('onChangeCurrentPage.0.0', 2);
});

it('should emit on previous button', async () => {
  const wrapper = mount(ToolbarPagination, {
    props: {
      ...props,
      currentPage: 7,
    },
  });
  await wrapper.findAll('button')[0].trigger('click');
  expect(wrapper.emitted()).toHaveProperty('onChangeCurrentPage.0.0', 6);
});
