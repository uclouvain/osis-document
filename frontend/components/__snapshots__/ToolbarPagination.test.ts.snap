// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should mount 1`] = `"<div class=\\"btn-group\\"><button class=\\"btn btn-default disabled\\" type=\\"button\\"><span class=\\"fas fa-angle-up\\"></span></button><a class=\\"btn btn-default disabled\\" role=\\"button\\">editor.pagination</a><button class=\\"btn btn-default\\" type=\\"button\\"><span class=\\"fas fa-angle-down\\"></span></button></div>"`;
